torchvision/prototype/transforms/_utils.py

```diff
@@ -39,14 +39,27 @@ def get_image_dimensions(image: Union[PIL.Image.Image, torch.Tensor, features.Im
     return channels, height, width
 
 
-def has_any(sample: Any, *types: Type) -> bool:
+def _parse_types_or_checks(
+    types_or_checks: Tuple[Union[Type, Callable[[Any], bool]], ...]
+) -> List[Callable[[Any], bool]]:
+    return [
+        cast(Callable[[Any], bool], lambda obj, typ=type_or_check: isinstance(obj, typ))
+        if isinstance(type_or_check, type)
+        else type_or_check
+        for type_or_check in types_or_checks
+    ]
```

Comment:
Nit: This is correct but really densely written. I had to eyeball it for 2 minutes to understand that the if/else is after the loop. Might be worth refactoring with loops to make it clearer, even if that's less pythonic.

Reply:
Unsolicited opinion: I find it hard to read too. The heavy type annotations, the call to `cast()` and the use of `lambda` with a default value also impair readability IMHO.

Reply:
I agree, the lambda with annotations is hard to read. I think we can safely replace the `cast` with an ignore comment. Plus, we can simplify the function annotations a little if we accept `*types_or_checks`.

However, I don't have a more readable way to create callables in a loop other than the default value. Happy to take suggestions here.